src/app/hooks/useRbfTransactionData.ts

```diff
@@ -0,0 +1,274 @@
+      const fee = microStxToStx(transaction.fee);
+      const txRaw: string = await getRawTransaction(transaction.txid, network);
+      const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+      const [slow, medium, high] = await estimateTransaction(
+        unsignedTx.payload,
+        undefined,
+        selectedNetwork,
+      );
+
+      let minimumFee = fee.multipliedBy(1.25);
+      if (!Number.isSafeInteger(minimumFee)) {
+        // round up the fee to the nearest integer
+        minimumFee = microStxToStx(Math.ceil(stxToMicrostacks(minimumFee).toNumber()));
+      }
+
+      const currentMicrostacksFee = stxToMicrostacks(fee);
+      let feePresets: RbfRecommendedFees = {};
+      const mediumFee = medium.fee;
+      const highFee = high.fee;
+      let higherFee = highFee * 1.25;
+      const highestFee = currentMicrostacksFee.multipliedBy(1.5).toNumber();
+
+      if (currentMicrostacksFee.lt(BigNumber(mediumFee))) {
+        feePresets = await constructRecommendedFees(
+          'medium',
+          mediumFee,
+          'high',
+          highFee,
+          stxAvailableBalance,
+          feeMultipliers?.thresholdHighStacksFee,
+        );
+      } else if (
+        currentMicrostacksFee.gt(BigNumber(mediumFee)) &&
+        currentMicrostacksFee.lt(BigNumber(highFee))
+      ) {
+        feePresets = await constructRecommendedFees(
+          'high',
+          highFee,
+          'higher',
+          higherFee,
+          stxAvailableBalance,
+          feeMultipliers?.thresholdHighStacksFee,
+        );
+      } else {
+        higherFee = currentMicrostacksFee.multipliedBy(1.25).toNumber();
+
+        feePresets = await constructRecommendedFees(
+          'higher',
+          higherFee,
+          'highest',
+          highestFee,
+          stxAvailableBalance,
+          feeMultipliers?.thresholdHighStacksFee,
+        );
+      }
+
+      setRbfData({
+        rbfTransaction: undefined,
+        rbfTxSummary: {
+          currentFee: fee.toNumber(),
+          currentFeeRate: fee.toNumber(),
+          minimumRbfFee: minimumFee.toNumber(),
+          minimumRbfFeeRate: minimumFee.toNumber(),
+        },
+        rbfRecommendedFees: sortFees(feePresets),
+        mempoolFees: {
+          fastestFee: microStxToStx(high.fee).toNumber(),
+          halfHourFee: microStxToStx(medium.fee).toNumber(),
+          hourFee: microStxToStx(slow.fee).toNumber(),
+          economyFee: microStxToStx(slow.fee).toNumber(),
+          minimumFee: microStxToStx(slow.fee).toNumber(),
+        },
+      });
+    } catch (err: any) {
+      console.error(err);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [transaction, network, selectedNetwork, feeMultipliers, stxAvailableBalance]);
+
+  const fetchRbfData = useCallback(async () => {
+    if (!selectedAccount || !transaction) {
+      setIsLoading(false);
+      return;
+    }
+
+    if (!isBtcTransaction(transaction)) {
+      return fetchStxData();
+    }
```

Comment:
it kinda feels like this should be two separate hooks. 

you end up loading more code than you need to, like btcClient (when doing stx rbf) and selectedNetwork, network, stxAvailableBalance (when doing btc rbf).

did you consider splitting the screens/speedUpTransaction/index.tsx into btc and stx versions? I saw you split it up on the component level instead

Reply:
that's a valid point.

regarding separate hooks - my suggestion here is to see how it'll look like when we move the helper functions and some of the STX logic to the core repo and then decide.

and regarding separate screens - not sure if it's necessary since they share quite a lot of logic and UI currently. if we make the separate hooks then we probably should separate the screens as well

Reply:
> if we make the separate hooks then we probably should separate the screens as well

yeah that would make sense

> regarding separate hooks - my suggestion here is to see how it'll look like when we move the helper functions and some of the STX logic to the core repo and then decide.

OK